### PR TITLE
Change k8s ServiceTypes to be changeable

### PR DESCRIPTION
**Pull request by withsmilo**
When creating a KubernetesManager object, you could define own ServiceTypes. For example,
```python
cm = KubernetesContainerManager(
         ....,
         service_types={
             'redis': 'NodePort',
             'management': 'LoadBalancer',
             'query': 'LoadBalancer',
             'query-rpc': 'ClusterIP',
             'metric': 'LoadBalancer'})
```
* related issue : https://github.com/ucbrise/clipper/issues/597

### CODE REVIEW

**Comment by simon-mo** [comment]
@withsmilo  Thanks for this patch. However, I think we also need to patch how `connect()` works because it currently assumes `NodePort` like line:
https://github.com/ucbrise/clipper/blob/227c8e46e5a801e9f94e7e42a5bf0b693a5f9919/clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py#L350

File: clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py
Comment (rkooo567) [inline clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py]: Why don't we make this part as a function and add some tests?

Comment (withsmilo) [reply clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py]: @rkooo567 , OK. Let me refactor this part.

**Comment by withsmilo** [comment]
@simon-mo , I will try to add something to need.

File: clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py
Comment (withsmilo) [reply clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py]: @rkooo567 , What tests are we needed here in additional?

Comment (rkooo567) [reply clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py]: I thought just a simple unit test? Like if this function raises Exception correctly when wrong input is given and etc. I could find that the current CI fails on line 200 (when you raise `ClipperException`), so it can be good if we can write some unit tests inside one of integration test files.

Comment (wcwang07) [reply clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py]: @withsmilo @rkooo567 can we add

def set_clipper_svc_types(self, svctype, svcname):
        for key in self.OFFICIAL_K8S_SERVICE_TYPE:
            if(svctype.lower()==key.lower()):
                self.DEFAULT_CLIPPER_SERVICE_TYPES[svcname]=key

Comment (withsmilo) [reply clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py]: @rkooo567 @wcwang07 
Sorry about late reply. I refactored my PR entirely.

File: clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py
Comment (rkooo567) [inline clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py:429]: Why don't we read service_types from our Kubernetes definitions (yaml file) in the `connect` function? Otherwise we should always specify the correct `self.service_types` whenever we initiate `KubernetesContainerManager`. 

It shouldn't matter when we initiate `KubernetesContainerManager` for the first time for `start_clipper`. But, let's say we already started clipper and just wanted to connect to the cluster using `ClipperConnection(KubernetesContainerManager()).connect()`. If you forget to write `service_types`, it will be set to be None and this part might be broken.

Comment (withsmilo) [reply clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py:429]: @rkooo567 You're right. Thank you for your kind advice. I will fix it as soon as possible.

Comment (withsmilo) [reply clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py:429]: @rkooo567 : According to your advice, I added a new mechanism to read/write `service_types` from/to a yaml file. You can connect to the executing Clipper cluster easily with `service_types`(None), which will be loaded automatically from a yaml file saved when we started it. If you call 'stop_all()', the yaml file will be removed.

Comment (rkooo567) [reply clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py:429]: Thank you! Everything looks good to me :)

**Comment by withsmilo** [comment]
@simon-mo : Please review this PR. Can we use 'LoadBalancer' service type with Travis CI?

**Comment by rkooo567** [comment]
This PR looks good to me. But since it is not a simple PR, I will wait for @simon-mo's approval.

**Review by simon-mo — APPROVED** [review]
lgtm

**Pull request merged**
